drivers/pwm/pwm_stm32.c

```diff
@@ -276,6 +281,127 @@ static int pwm_stm32_init(const struct device *dev)
 		return r;
 	}
 
+	/* configure pinmux */
+	if (cfg->pinctrl_len != 0U) {
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_pinctrl)
+		/* apply F1 series remaps */
+		int remap;
+
+		remap = stm32_dt_pinctrl_remap_check(cfg->pinctrl,
+						     cfg->pinctrl_len);
+		if (remap < 0) {
+			LOG_ERR("pinctrl remap check failed (%d)", remap);
+			return remap;
+		}
+
+		LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_AFIO);
+
+		switch ((uint32_t)cfg->timer) {
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(timers1), okay)
+		case DT_REG_ADDR(DT_NODELABEL(timers1)):
+			if (remap == REMAP_FULL) {
+				LL_GPIO_AF_EnableRemap_TIM1();
+			} else if (remap == REMAP_1) {
+				LL_GPIO_AF_RemapPartial_TIM1();
+			}
+			break;
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(timers2), okay)
+		case DT_REG_ADDR(DT_NODELABEL(timers2)):
+			if (remap == REMAP_FULL) {
+				LL_GPIO_AF_EnableRemap_TIM2();
+			} else if (remap == REMAP_1) {
+				LL_GPIO_AF_RemapPartial1_TIM2();
+			} else if (remap == REMAP_2) {
+				LL_GPIO_AF_RemapPartial2_TIM2();
+			}
+			break;
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(timers3), okay)
+		case DT_REG_ADDR(DT_NODELABEL(timers3)):
+			if (remap == REMAP_FULL) {
+				LL_GPIO_AF_EnableRemap_TIM3();
+			} else if (remap == REMAP_1) {
+				LL_GPIO_AF_RemapPartial_TIM3();
+			}
+			break;
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(timers4), okay)
+		case DT_REG_ADDR(DT_NODELABEL(timers4)):
+			if (remap == REMAP_FULL) {
+				LL_GPIO_AF_EnableRemap_TIM4();
+			}
+			break;
```

Comment:
Regarding Timer 5, the only available remap function is `LL_GPIO_AF_EnableRemap_TIM5CH4`. As stated in the summary, I'm not sure how to handle a channel dependent remap. Timer 8 has no remap options, at least on LL API.

Reply:
Let's keep it as is for now. If need arises, the usage will become clearer.

Reply:
Oups sorry, I forgot the point you mentioned in the summary, after the review of the 44 commits.
Ok for TIM8.
About TIM5 remap, even if the channel pin cannot really be remapped, it can still be disabled by setting bitfield TIM5CH4_IREMAP
![image](https://user-images.githubusercontent.com/50730894/95062097-c74a6980-06fc-11eb-8ccc-8d87d6dfbeeb.png)
So I would say it is safer to call LL_GPIO_AF_DisableRemap_TIM5CH4() to make sure remap TIM5_CH4 is connected to PA3.
```
#if DT_NODE_HAS_STATUS(DT_NODELABEL(timers5), okay)
		case DT_REG_ADDR(DT_NODELABEL(timers5)):
			LL_GPIO_AF_DisableRemap_TIM5CH4();
			break;
#endif`
```
But it seems that generated pinctrl files fot F1 doesn't contains anything about TIM5.

Reply:
ST database does not contain anything for TIM5. As the remap is for internal routing, PWM will never need such remap anyway. Regarding the call to DisableRemap, should we call it when `NO_REMAP` is found? If yes, we need to update all other cases as it is not handled. UART driver will also need to be updated to handle this case.

Reply:
> Regarding the call to DisableRemap, should we call it when `NO_REMAP` is found? If yes, we need to update all other cases as it is not handled. UART driver will also need to be updated to handle this case.

I think so. Otherwise we rely on hypothesis that nobody changes this (we remain on default value). And maybe this hypothesis is valid today, but maybe not on the future.

Reply:
@ABOSTM fixed

Reply:
> UART driver will also need to be updated to handle this case.

I'm pushing a fix for this one

Reply:
@ABOSTM can you review latest changeset?